src/Components/Components/src/Properties/ILLink.Substitutions.xml

```diff
@@ -1,17 +1,11 @@
 <linker>
-  <assembly fullname="Microsoft.AspNetCore.Components" >
-    <!-- HotReload will not be available in a trimmed app. We'll attempt to aggressively remove all references to it. -->
-    <type fullname="Microsoft.AspNetCore.Components.RenderTree.Renderer">
-      <!-- Renderer.IsHotReloading will always return false in a trimmed app. -->
-      <method signature="System.Boolean get_IsHotReloading()" body="stub" value="false" />
-      <method signature="System.Void RenderRootComponentsOnHotReload()" body="remove" />
-      <method signature="System.Void InitializeHotReload(System.IServiceProvider)" body="stub" />
-      <method signature="System.Void CaptureRootComponentForHotReload(Microsoft.AspNetCore.Components.ParameterView,Microsoft.AspNetCore.Components.Rendering.ComponentState)" body="stub" />
-      <method signature="System.Void DisposeForHotReload()" body="stub" />
-    </type>
+  <assembly fullname="Microsoft.AspNetCore.Components">
     <!-- Avoid any overhead in RenderHandle.IsHotReloading by aggressively trimming it -->
     <type fullname="Microsoft.AspNetCore.Components.RenderHandle">
        <method signature="System.Boolean get_IsHotReloading()" body="stub" value="false" />
```

Comment:
I think we'd specifically wanted to short-circuit this one quickly since it appears as part of the rendering loop. Seems easy enough to keep it

Reply:
I'm not sure I follow this reasoning. The body of this method will already be trimmed to just `return false` since 

> public bool IsHotReloading => HotReloadFeature.IsSupported && (_renderer?.IsHotReloading ?? false);

will evaluate to `false && (_renderer?.IsHotReloading ?? false)`, which is always `false`.

Further, the only call I see to `RenderHandle.IsHotReloading` is here:

https://github.com/dotnet/aspnetcore/blob/07c6a62587fb147cefbd5f84c5a308b5e087cb9c/src/Components/Components/src/ComponentBase.cs#L98-L105

Which comes after a `virtual ShouldRender()` call.

So I'm not sure this entry is adding any benefit.

Reply:
This is how it looks like in the absence of the explicit trim:

```C#
public bool IsHotReloading
{
	get
	{
		if (HotReloadFeature.IsSupported)
		{
		}
		return false;
	}
}
```

vs
```C#
public bool IsHotReloading => false;
```

`StateHasChanged` is part of the render-loop so it would be ideal to have it execute additional instructions it doesn't really need to.

Reply:
cc @marek-safar - here's more feedback on the trimmer leaving the call to an unnecessary method after trimming. See https://github.com/mono/linker/issues/1113

Reply:
>  is part of the render-loop so it would be ideal to have it execute additional instructions it doesn't really need to.

@pranavkm the IL does not fully represent executed code. In this case, the whole method will be skipped by AOT or interpreter. We'll eventually optimize the IL as well but it's nice to have at this point because it does not affect the speed only size and the savings are small.

Reply:
Cool. I didn't realise the interpreter had the ability to do things like skip empty methods. I'd be interested to know if there's a particular way we should be able to reason about what optimizations would or wouldn't happen in interpreted and AOT modes.

In any case if there's no runtime cost to having a call to an empty method that's great, and it will be nice to remove this special case from the linker config!